pilot/pkg/model/push_context.go

```diff
@@ -1812,3 +1812,33 @@ func (ps *PushContext) NetworkGatewaysByNetwork(network string) []*Gateway {
 func (ps *PushContext) QuotaSpecByDestination(instance *ServiceInstance) []Config {
 	return filterQuotaSpecsByDestination(instance, ps.QuotaSpecBinding, ps.QuotaSpec)
 }
+
+// BestEffortInferServiceMTLSMode infers the mTLS mode for the service + port from all authentication
+// policies (both alpha and beta) in the system. The function always returns MTLSUnknown for external service.
+// The resulst is a best effort. It is because the PeerAuthentication is workload-based, this function is unable
+// to compute the correct service mTLS mode without knowing service to workload binding. For now, this
+// function uses only mesh and namespace level PeerAuthentication and ignore workload & port level policies.
+// This function is used to give a hint for auto-mTLS configuration on client side.
+func (ps *PushContext) BestEffortInferServiceMTLSMode(service *Service, port *Port) MutualTLSMode {
+	if service.MeshExternal {
+		// Only need the authentication MTLS mode when service is not external.
+		return MTLSUnknown
```

Comment:
should be MTLSDisable

Reply:
This is to keep the same behavior as before. Maybe keep it as is to avoid more regression.